realm/realm-library/src/main/java/io/realm/RealmDictionary.java

```diff
@@ -109,44 +110,70 @@ public RealmDictionary(RealmDictionary<V> dictionary) {
     private static <V> ManagedMapStrategy<String, V> getStrategy(Class<V> valueClass, BaseRealm baseRealm, OsMap osMap) {
         if (isClassForRealmModel(valueClass)) {
             ClassContainer classContainer = new ClassContainer(valueClass, null);
-            RealmModelValueOperator<RealmModel> realmModelValueOperator = new RealmModelValueOperator<>(baseRealm, osMap, classContainer);
-            ManagedMapManager<String, RealmModel> dictionaryManager = new DictionaryManager<>(realmModelValueOperator);
+            RealmModelValueOperator<String, RealmModel> realmModelValueOperator = new RealmModelValueOperator<>(baseRealm, osMap, classContainer);
+            ManagedMapManager<String, RealmModel> dictionaryManager = new DictionaryManager<>(realmModelValueOperator, classContainer);
 
             //noinspection unchecked
             return (ManagedMapStrategy<String, V>) new ManagedMapStrategy<>(dictionaryManager);
         }
-        return getStrategy(valueClass.getCanonicalName(), baseRealm, osMap);
+
+        DictionaryManager<V> manager = getManager(valueClass, baseRealm, osMap);
+        return new ManagedMapStrategy<>(manager);
     }
 
     private static <V> ManagedMapStrategy<String, V> getStrategy(String valueClass, BaseRealm baseRealm, OsMap osMap) {
         DictionaryManager<V> manager = getManager(valueClass, baseRealm, osMap);
         return new ManagedMapStrategy<>(manager);
     }
 
-    private static <V> DictionaryManager<V> getManager(String valueClass, BaseRealm baseRealm, OsMap osMap) {
-        // TODO: add other types when ready
+    private static <K, V> DictionaryManager<V> getManager(Class<V> valueClass, BaseRealm baseRealm, OsMap osMap) {
+        ClassContainer classContainer = new ClassContainer(valueClass, null);
+
+        MapValueOperator<?, ?> mapValueOperator;
```

Comment:
```suggestion
        MapValueOperator<K, ?> mapValueOperator;
```
The K type can be kept.